### ISSUE TITLE
[PSR-6] Decoupling CacheItem from CacheItemPool

### ISSUE THREAD

**Issue by rindeal** [php-fig/fig-standards#609]
As already pointed out in #389, method `exists()` is the only method in `CacheItemInterface` which requires to know which `CacheItemPool` the item belongs to.

When trying to figure out, why `exists()` exists at all, I stumbled upon [this article](http://blog.ircmaxell.com/2014/10/a-followup-to-open-letter-to-php-fig.html), which explains that the reason is memory efficiency for someone who just needs to peek into the cache to see if the item exists without caring what it's value is. To avoid race conditions the caller will also likely not use any other method provided in `CacheItemInterface`, unless he calls `getItem()` again.

To sum it up, I see no reason why `exists()` shouldn't be moved to `CacheItemPoolInterface`.

**Comment by tedivm** [comment]
This is an excellent point, but it should be brought up on the mailing list.

**Comment by rindeal** [comment]
Thanks. [Here is the link to mailing list conversation](https://groups.google.com/forum/#!topic/php-fig/NKaQhF98TVI)

**Comment by dragoonis** [comment]
Closing this in favour of #612
